### PR TITLE
Fixing execution of `render_default_user_ssl_exports` before etcd is installed

### PR DESCRIPTION
**Pull request by joedborg**
Re. https://bugs.launchpad.net/charm-etcd/+bug/1861483

Have tested this with CK stable and edge.  Have tested an etcd snap upgrade as well as etcd charm upgrade.

Will hold to test with the exact failure case on the bug report.

### CODE REVIEW

**Comment by joedborg** [comment]
[Install hook regularly fails due to etcd version check before etcd is installed](https://trello.com/c/k9g6J5zV/1433-install-hook-regularly-fails-due-to-etcd-version-check-before-etcd-is-installed)

File: reactive/etcd.py [hunk 1]
```diff
@@ -308,7 +309,6 @@ def snap_install():
     snap.install('core')
     if channel:
         snap.install('etcd', channel=channel, classic=False)
-    remove_state('etcd.pillowmints')
```
Comment (Cynerva) [inline reactive/etcd.py:311]: Why did you remove this?

File: reactive/etcd.py [hunk 2]
```diff
@@ -329,6 +329,8 @@ def install_etcd():
     if channel:
         snap.install('etcd', channel=channel, classic=False)
 
+    set_state('etcd.installed')
```
Comment (Cynerva) [inline reactive/etcd.py]: Setting `etcd.installed` here will confuse the charm and make it think that a migration to snaps needs to occur, won't it?

`etcd.installed` is a legacy state that comes from the old days when we delivered etcd via charm resource, not snap. The charm still uses that state as a signal that the upgrade from pre-snap -> snap has not occurred. See for example this handler:

https://github.com/charmed-kubernetes/layer-etcd/blob/0e450a360598f4e4ac247e343e05b9aba9e1f04f/reactive/etcd.py#L83-L85

File: reactive/etcd.py (hunk 1 above)
Comment (joedborg) [reply reactive/etcd.py:311]: I have replaced `etcd.pillowmints` with `etcd.ssl.exported` as that's all it seemed to do anymore.

Comment (Cynerva) [reply reactive/etcd.py:311]: Right, but this wasn't replaced with `remove_state('etcd.ssl.exported')`, it was just removed. If we don't clear the `etcd.ssl.exported` flag here, then `render_default_user_ssl_exports` won't be re-run after an upgrade from etcd2 -> etcd3 will it?

File: reactive/etcd.py [hunk 3]
```diff
@@ -329,6 +329,8 @@ def install_etcd():
     if channel:
         snap.install('etcd', channel=channel, classic=False)
 
+    set_state('snap.installed.etcd')
```
Comment (Cynerva) [inline reactive/etcd.py]: I don't think this is necessary given that `snap.install('etcd', ...)` sets the same state here: https://github.com/stub42/layer-snap/blob/9f8b9a03f57b528d200d037789db910ca79698b9/lib/charms/layer/snap.py#L58

File: reactive/etcd.py (hunk 1 above)
Comment (joedborg) [reply reactive/etcd.py:311]: I think it'll be cleared by L148-149 `@hook('upgrade-charm')` - `def remove_states():`.  I thought this route had to be taken if you were going deb->snap?

File: reactive/etcd.py (hunk 3 above)
Comment (joedborg) [reply reactive/etcd.py]: Ah, thanks!

File: reactive/etcd.py (hunk 1 above)
Comment (Cynerva) [reply reactive/etcd.py:311]: Nah, I'm talking about snap->snap upgrade from etcd 2.3 to 3.x, where the user does this:

```
juju config etcd channel=3.0/stable
```

It'll cause `snap_install` to run again, but with no `upgrade-charm` event.

Comment (joedborg) [reply reactive/etcd.py:311]: Ah yeah, I see what you mean.  Fixed!  Thanks.

**Comment by Cynerva** [comment]
Thanks. LGTM :+1:
